components/org.wso2.carbon.identity.oauth2.grant.organizationswitch/src/main/java/org/wso2/carbon/identity/oauth2/grant/organizationswitch/OrganizationSwitchGrant.java

```diff
@@ -134,6 +144,26 @@ public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx) throws Id
         return true;
     }
 
+    private boolean isActiveOrganization(String organizationId) throws IdentityOAuth2Exception {
+
+        try {
+            boolean organizationExistById = OrganizationSwitchGrantDataHolder.getInstance().getOrganizationManager()
+                    .isOrganizationExistById(organizationId);
+            if (!organizationExistById) {
+                throw new IdentityOAuth2Exception(OAuth2ErrorCodes.INVALID_REQUEST,
+                        "Invalid organization id: " + organizationId);
+            }
+            String tenantDomain = getTenantDomainFromOrgId(organizationId);
+            TenantManager tenantManager =
+                    OrganizationSwitchGrantDataHolder.getInstance().getRealmService().getTenantManager();
+            return tenantManager.isTenantActive(IdentityTenantUtil.getTenantId(tenantDomain));
+        } catch (OrganizationManagementException e) {
+            throw new RuntimeException(e);
```

Comment:
```suggestion
                        throw new IdentityOAuth2Exception("Error while checking organization exist with ID: " + organizationId , e);
```

Reply:
fixed in [530ec61](https://github.com/wso2-extensions/identity-oauth2-grant-organization-switch/pull/34/commits/530ec61b0910f231a3c19565996e55b85f5bc2f8)